scipy/stats/_continuous_distns.py

```diff
@@ -7645,11 +7645,17 @@ def _logpdf(self, x, c):
         return np.log(c) + _norm_logpdf(x) + (c-1)*_norm_logcdf(-x)
 
     def _cdf(self, x, c):
-        return 1.0-_norm_cdf(-x)**(c*1.0)
+        return -sc.expm1(c * _norm_logcdf(-x))
 
     def _ppf(self, q, c):
         return -_norm_ppf(pow(1.0 - q, 1.0 / c))
 
+    def _sf(self, x, c):
+        return np.exp(c * _norm_logcdf(-x))
+
+    def _isf(self, q, c):
+        return - _norm_ppf(q**(1 / c))
```

Comment:
Style nit: no space after a unary minus.
```suggestion
        return -_norm_ppf(q**(1 / c))
```